### PR TITLE
introduce gh-action-runner deployment

### PR DESCRIPTION
**Pull request by eslutsky**
deploy gh-action self-hosted runner VM on ovirt this is the  flow:

- [x]  ansible that creates a  Fedora based gh runner Template (with everything pre-installed)
  -  based on a fedora-46 OS  
  -  packages and kind preinstalled 
  - gh-runner agent with forklift token pre-installed
- [x]  ansible that  creates a  VM from the template 
   - once the VM is created it will join the forklift repo as a runner
- [x] cleanup code

### CODE REVIEW

File: cluster/gh-action-runner/gh-action-runner-template-create.yml [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+          state: stopped
+
+      - ovirt_vm_info:
+          pattern: name=fedora_test_vm and cluster="{{ovirt_cluster_name}}"
+          auth: "{{ovirt_auth}}"
+          follow: ['disk_attachments']
+        register: result
+     
+      - debug:
+          msg: "{{ result  }}"
```
Comment (mnecas) [inline cluster/gh-action-runner/gh-action-runner-template-create.yml:82]: Remove extra space after `result`

File: cluster/gh-action-runner/gh-action-runner-template-create.yml [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+          async: 1
+          poll: 0
+        - name: wait for server to come back after reboot
+          wait_for_connection:
+        delegate_to: test-vm
+
+      - name: install pre-req
+        block:
+        - name: install-dev-tools
+          dnf: name="@Development tools" state=present
```
Comment (mnecas) [inline cluster/gh-action-runner/gh-action-runner-template-create.yml:187]: Please split the string to dict items

File: cluster/gh-action-runner/gh-action-runner-template-create.yml [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+            tar -C /usr/local -xzf go1.20.4.linux-amd64.tar.gz
+
+        - name: install kubectl
+          shell: |
+            curl -L "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" -o /usr/local/bin/kubectl
+            chmod +x /usr/local/bin/kubectl
+
+        - name: install k9s
+          shell: |
+            curl -L https://github.com/derailed/k9s/releases/download/v0.27.4/k9s_Linux_amd64.tar.gz -o /tmp/k9s.tar.gz
```
Comment (mnecas) [inline cluster/gh-action-runner/gh-action-runner-template-create.yml:230]: Hmm would probably also move this to some variable so we don't need to go through code to know which version do we install

File: cluster/gh-action-runner/gh-action-runner-template-create.yml [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+        ping:
+        delegate_to: test-vm
+
+      - name: resize the root partition 
+        shell: |
+          growpart  /dev/vda 1
+          resize2fs /dev/vda1
+        delegate_to: test-vm
+
+      - name: upgrade to fedora-36
```
Comment (mnecas) [inline cluster/gh-action-runner/gh-action-runner-template-create.yml:168]: Why do we need to upgrade to fedora 36?
What is the base OS version and why can't we install the 36 originally?

**Review by mnecas — CHANGES_REQUESTED** [review]

File: cluster/gh-action-runner/gh-action-runner-template-create.yml [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+          cluster: "{{ovirt_cluster_name}}"
+          auth: "{{ovirt_auth}}"
+          memory: 14GiB
+          cpu_cores: 12
+          state: running
+          bios_type: "i440fx_sea_bios"
+          nics:
+            - name: eth0
+              profile_name: "{{rhv_network}}"
+              mac_address: "{{test_vm_mac_address | default(omit)}}"
```
[the remark this answers is not in the document]
Comment (eslutsky) [reply cluster/gh-action-runner/gh-action-runner-template-create.yml]: its not set , so its  omitted in our case, 
but its useful when deploying on environment  where we want to control the IP address (throughout a pre-define mac-address reservation)

File: cluster/gh-action-runner/gh-action-runner-template-create.yml [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+      engine_fqdn: "{{ ovirt_engine_url }}"
+      engine_user: "{{ ovirt_engine_username }}"
+      engine_password: "{{ ovirt_engine_password }}"
+      docker_ce_repo_url: https://download.docker.com/linux/fedora
+      docker_ce_repo_file_url: "{{ docker_ce_repo_url }}/docker-ce.repo"
+      docker_ce_gpg_key: "{{ docker_ce_repo_url }}/gpg"
+      docker_ce_edge_repo_url: "{{ docker_ce_repo_url }}/{{ ansible_distribution_major_version }}/$basearch/edge"
+    collections:
+      - ovirt.ovirt
+      - ansible.posix
```
[the remark this answers is not in the document]
Comment (eslutsky) [reply cluster/gh-action-runner/gh-action-runner-template-create.yml]: yea its become pretty long .  but i think we do this in different refactor PR.

**Review by ahadas — APPROVED** [review]

File: cluster/gh-action-runner/gh-action-runner-template-create.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (ahadas) [reply cluster/gh-action-runner/gh-action-runner-template-create.yml]: > but i think we do this in different refactor PR.

ack

**Pull request merged**